.github/scripts/publish-script.sh

```diff
@@ -21,4 +21,4 @@ cp ./LICENSE ./lib
 cp ./README.md ./lib
 
 # publish
-pnpm publish ./lib --ignore-scripts
\ No newline at end of file
+pnpm publish ./lib --ignore-scripts --no-git-checks
```

Comment:
why this flag?

Reply:
To answer for why we might have uncommitted change at ci level: 
consider the above pkg version (1.0.1-17.g9bc78dc)
1. we generate a temp version for snapshot build here is the formula: 
    1.0.1-<number of commits this branch is ahead of main>.<commit hash>
2. then we set that in pkg.json using `npm version <GENERATED_VERSION>` in [this bash script](https://github.com/qlik-oss/nebula-table-utils/blob/ebb8b0cd1573f69c241234211743be1608fa403d/.github/scripts/generate-pkg-version.sh#L49) which runs as part of dev build workflow. 
3. [pnpm prevents publish](https://stackoverflow.com/questions/62428993/error-unclean-working-tree-commit-or-stash-changes-first) if you have some uncommitted change
4. so we skip git checks for that

Reply:
do we lose anything from not doing the checks? no
do we know that everything is fine before publishing? yes, the only change is the version in package.json file which we need for publishing

basically nothing from snapshot publishes might get into the end users hand -> so we are safe with them

Reply:
Ok then I think it is fine :)